### PR TITLE
[Monitor] Add two missing "by subscription" operations for completeness and compliance

### PR DESCRIPTION
**Pull request by gucalder**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/microsoft.insights/stable/2015-04-01/autoscale_API.json [hunk 1]
```diff
@@ -834,6 +834,180 @@
...
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request for a list of autoscale settings",
+            "schema": {
+              "$ref": "#/definitions/AutoscaleSettingResourceCollection"
+            },
+            "examples": {
```
[the remark this answers is not in the document]
Comment (gucalder) [reply specification/monitor/resource-manager/microsoft.insights/stable/2015-04-01/autoscale_API.json]: "Examples" nodes removed. The actual examples are already in the x-ms-examples referenced files.

**Comment by gucalder** [comment]
Last commit addresses issue #3293

**Comment by dsgouda** [comment]
@gucalder Some of the added examples seem to fail validation. Please take look at the logs [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/395287731) and fix the issues

**Comment by gucalder** [comment]
@dsgouda To be honest, I do not understand the errors in metricAlert (I do not own this one)
The only thing that comes to my mind is:
Should I remove the second member of the odatatype enumeration for MetricAlertCriteria?
,
            "Microsoft.Azure.Monitor.DefaultCriteria" 

Because it is not defined and/or used anywhere else.

**Comment by dsgouda** [comment]
@gucalder I think the API owners should be fixing it. Will open an issue regarding this.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@gucalder as a best practice please make changes in a branch other than `master` for opening the PRs. Keeps the commit history clean and would be easier for you too

**Pull request merged**